pkg/kubelet/kubelet.go

```diff
@@ -1722,12 +1752,41 @@ func (kl *Kubelet) makePodDataDirs(pod *api.Pod) error {
 	return nil
 }
 
+// syncPod is the transaction script for the sync of a single pod.
+//
+// Arguments:
+//
+// pod - the pod to sync
+// mirrorPod - the mirror pod for the pod to sync, if it is a static pod
+// podStatus - the current status (TODO: always from the status manager?)
+// updateType - the type of update (CREATE,ADD,DELETE,RECONCILE)
+//
+// The workflow is:
```

Comment:
I like numbers for things with a bunch of steps, or that are especially important.  IMO it is important to get the behaviors of an important method like this into the comment.  I want to give as much information as possible to reason about qualitatively without having to dive into the kube.  I am cool with replacing the numbers with bullets if people really don't like it, but I have to call out this type of criticism as being equally applicable to any godoc -- if you change the behavior of a function significantly odds are you will be updating the doc for the func.